### ISSUE TITLE
Feature: Keyword-based antispam

### ISSUE THREAD

**Issue by 1nsecure** [WhisperSystems/Signal-Android#6727]
It would be nice to have feature to automatically silence notification and/or block contact, based on specified keywords.
We all don't like spammers ;)

**Comment by p4nci** [comment]
I know it's not automated, but you can block 'spammers' already:
[How do I block a number](https://support.whispersystems.org/hc/en-us/articles/212535898-How-do-I-block-a-number-on-Signal-Android-)

**Comment by 1nsecure** [comment]
Right, but automation here is kinda useful to keep annoying people away :)

**Comment by 1nsecure** [comment]
BTW, https://www.bountysource.com/issues/46142714-feature-keyword-based-antispam

**Comment by paride** [comment]
The only reasonable way to implement this is to have an option to block messages from unknown contacts.

**Comment by 1nsecure** [comment]
I'm pretty sure, that Signal can "mute" contacts, based on sms's content. Like, this is not physically impossible :)

**Comment by 1nsecure** [comment]
If anyone else thinks that this feature is useless or impossible to implement, feel free to close the issue.

**Comment by annabeth060** [comment]
You want to user enter some words so that received messages containing that words never appear in the app. Did I understand it correctly?

**Comment by 1nsecure** [comment]
Yes. Receive them quietly or just delete them.

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
